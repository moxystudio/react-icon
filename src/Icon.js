import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Svg from '@moxy/react-svg';

export const Icon = forwardRef(({ className, ...rest }, ref) => {
    const finalProps = {
        ...rest,
        className: classNames('react-icon', className),
    };

    return <Svg ref={ ref } { ...finalProps } />;
});

Icon.propTypes = {
    svg: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({ then: PropTypes.func.isRequired })]).isRequired,
    className: PropTypes.string,
};

export default Icon;
