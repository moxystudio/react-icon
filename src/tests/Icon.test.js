import React from 'react';
import { render } from '@testing-library/react';

import { Icon } from '../Icon';

const svg = `<svg height="100" width="100"> 
                <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3"/>
            </svg>`;

const defaultProps = {
    svg,
};

const renderWithProps = (props = {}) => render(<Icon { ...defaultProps } { ...props } />);

describe('Icon', () => {
    it('should render correctly', () => {
        const { asFragment } = renderWithProps();

        expect(asFragment()).toMatchSnapshot();
    });
});
