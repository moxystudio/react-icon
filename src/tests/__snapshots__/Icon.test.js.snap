// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<DocumentFragment>
  <svg
    class="react-svg react-icon"
    height="100"
    width="100"
  >
     
                
    <circle
      cx="50"
      cy="50"
      r="40"
      stroke="black"
      stroke-width="3"
    />
    
            
  </svg>
</DocumentFragment>
`;
